controllers/topology/reconcile_state.go

```diff
@@ -347,6 +435,13 @@ func (r *ClusterReconciler) updateMachineDeployment(ctx context.Context, cluster
 		return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: currentMD.Object})
 	}
 
+	// Patch MachineHealthCheck for the MachineDeployment.
+	if desiredMD.MachineHealthCheck != nil || currentMD.MachineHealthCheck != nil {
+		if err := r.reconcileMachineHealthCheck(ctx, currentMD.MachineHealthCheck, desiredMD.MachineHealthCheck); err != nil {
```

Comment:
Should we set owner references in case we are creating the MHC (current = nil)
Also wondering if we can pass the list of owners and delegate to reconcileMachineHealthCheck retrieving them in case of create, so we avoid to duplicate gets calls